### PR TITLE
add testcase for gdImageCopyMerge

### PR DESCRIPTION
**Pull request by willson-chen**

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -37,6 +37,7 @@ if (BUILD_TEST)
 		gdimagecopy
 		gdimagecopyresampled
 		gdimagecopyrotated
+		gdimagecopymerge
```
Comment (vapier) [inline tests/CMakeLists.txt]: please keep it sorted

File: tests/Makefile.am [hunk 2]
```diff
@@ -33,6 +33,7 @@ include gdimageconvolution/Makemodule.am
 include gdimagecopy/Makemodule.am
 include gdimagecopyresampled/Makemodule.am
 include gdimagecopyrotated/Makemodule.am
+include gdimagecopymerge/Makemodule.am
```
Comment (vapier) [inline tests/Makefile.am]: please keep it sorted

File: tests/gdimagecopymerge/Makemodule.am [hunk 3]
```diff
@@ -0,0 +1,7 @@
+if HAVE_LIBPNG
+	libgd_test_programs += \
```
Comment (vapier) [inline tests/gdimagecopymerge/Makemodule.am]: don't indent the bodies in Makemodule.am files under tests/

File: tests/gdimagecopymerge/gdimagecopymerge.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
+#include "gd.h"
+#include "gdtest.h"
+
+int main()
+{
+	gdImagePtr dst = NULL;
+	gdImagePtr src = NULL;
+	//FILE *pngout = NULL;
```
Comment (vapier) [inline tests/gdimagecopymerge/gdimagecopymerge.c]: delete dead/commented code

File: tests/gdimagecopymerge/gdimagecopymerge.c [hunk 5]
```diff
@@ -0,0 +1,41 @@
+#include "gd.h"
```
Comment (vapier) [inline tests/gdimagecopymerge/gdimagecopymerge.c]: please add comment blocks to all new files that give a brief overview of the file and its goals/purposes.  i know our existing files haven't been great in this regard, but we should be doing it better with new ones.

Comment (willson-chen) [reply tests/gdimagecopymerge/gdimagecopymerge.c]: totally agree

File: tests/gdimagecopymerge/Makemodule.am (hunk 3 above)
Comment (willson-chen) [reply tests/gdimagecopymerge/Makemodule.am]: do you mind telling me why

Comment (vapier) [reply tests/gdimagecopymerge/Makemodule.am]: we haven't been doing it thus far and i don't think we need to start, and it makes it somewhat easier to diff cmake & autotool files

File: tests/gdimagecopymerge/Makemodule.am [hunk 6]
```diff
@@ -0,0 +1,7 @@
+if HAVE_LIBPNG
+libgd_test_programs += \
+	gdimagecopymerge/gdimagecopymerge
+endif
+
+EXTRA_DIST += \
```
Comment (vapier) [inline tests/gdimagecopymerge/Makemodule.am]: shouldn't this be including the new cmake file ?

Comment (willson-chen) [reply tests/gdimagecopymerge/Makemodule.am]: sorry... missed it

**Comment by willson-chen** [comment]
sorry... i miss the indent in `CMakeLists.txt`.

thx! :)

**Pull request merged**